### PR TITLE
Allowed assigning PR author in PRs only

### PR DESCRIPTION
**Pull request by smmr-dn**
## Changes

Only allowed assigning author in PR and disabled this CI item from running when the PR is merged (mentioned in this comment https://github.com/iTwin/iTwinUI/pull/2231#discussion_r1764014043)

## Testing

Confirmed that PR author is still assigned in PRs.

## Docs

N/A

### CODE REVIEW

File: .github/workflows/assign-PR-author.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Assign PR author
```
Comment (mayank99) [inline .github/workflows/assign-PR-author.yml:1]: I would recommend renaming this file to `PR.yml` and naming the workflow "PR" as well. In the future, we can add any other PR-related automation in this file too.

File: .github/workflows/assign-PR-author.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+on:
+  pull_request:
+    types: [opened, reopened, ready_for_review]
```
Comment (mayank99) [inline .github/workflows/assign-PR-author.yml:4]: I think the `types` is not necessary?

File: .github/workflows/assign-PR-author.yml (hunk 1 above)
Comment (smmr-dn) [reply .github/workflows/assign-PR-author.yml]: I have renamed both.

File: .github/workflows/assign-PR-author.yml (hunk 2 above)
Comment (smmr-dn) [reply .github/workflows/assign-PR-author.yml]: I have removed `types`.

**Review by mayank99 — APPROVED** [review]
LGTM

**Pull request merged**